src/scene_battle_rpg2k3.cpp

```diff
@@ -1318,10 +1370,18 @@ void Scene_Battle_Rpg2k3::Escape(bool force_allow) {
 
 	SetState(State_SelectActor);
 	ShowNotification(ToString(lcf::Data::terms.escape_failure));
+	SetWait(10, 30);
 }
 
 bool Scene_Battle_Rpg2k3::CheckWin() {
 	if (Game_Battle::CheckWin()) {
+		if (!win_wait_elapsed) {
```

Comment:
This and the corresponding lose logic looks incorrect.

We still call `Scene_Battler::Update()` every frame which tries to run the event interpreter in `Game_Battle::UpdateEvents()`. Since you haven't "finished" the battle, in theory it appears to me the battle could end, then an event runs while this wait is processing which revives an enemy, and then the battle continues.

That being said, this part of the battle scene logic for 2k and 2k3 needs a refactor. So I am ok accepting your change here as is and fixing this part later.

Reply:
yeah this will continue running the battle after a victory which will have side effects because it lies about the state.